### PR TITLE
Initial app setup and auth

### PR DESCRIPTION
**Pull request by zpawn**
Story https://app.shortcut.com/deskpro/story/86054/teamviewer-initial-app-setup-and-auth

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import { DeskproAppProvider } from "@deskpro/app-sdk";
+import { StoreProvider } from "./context/StoreProvider";
 import { Main } from "./pages/Main";
 import "./App.css";
 
+import "iframe-resizer/js/iframeResizer.contentWindow.js";
```
Comment (AshleyDawson) [inline src/App.tsx:6]: You don't need to do this as it's included here: https://github.com/DeskproApps/teamviewer/blob/master/index.html#L7

File: src/hooks/useGlobalSignIn.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        const url = new URL(`https://webapi.teamviewer.com/api/v1/oauth2/token`);
+
+        const requestOptions: RequestInit = {
+            method: "POST",
+            body: new URLSearchParams({
+                grant_type: "authorization_code",
+                code: accessCode as string,
+                client_id: settings?.client_id as string,
+                client_secret: settings?.client_secret as string,
```
Comment (AshleyDawson) [inline src/hooks/useGlobalSignIn.ts:79]: Never include secrets via the frontend, this must be passed as a proxy placeholder. `client_secret` must be a "backend only" setting in the manifest too.

File: src/hooks/useGlobalSignIn.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+            client.setAdminSetting(JSON.stringify(tokens));
+
+            setIsLoading(false);
+        })();
+    }, [
+        accessCode,
+        callbackUrl,
+        settings?.client_id,
+        settings?.client_secret,
```
Comment (AshleyDawson) [inline src/hooks/useGlobalSignIn.ts:108]: Don't expose secrets in frontend - this should be a "backend only" setting

**Review by AshleyDawson — CHANGES_REQUESTED** [review]

File: src/hooks/useGlobalSignIn.ts (hunk 2 above)
Comment (zpawn) [reply src/hooks/useGlobalSignIn.ts:79]: @AshleyDawson I copied this approach from the Salesforce app. I specified that it `isBackendOnly`. Will the placeholder work in `adminGenericProxyFetch`?

Screenshot 2022-09-14 at 12 52 57

Comment (zpawn) [reply src/hooks/useGlobalSignIn.ts:79]: then it will have to be fixed in the Salesforce App too

Comment (AshleyDawson) [reply src/hooks/useGlobalSignIn.ts:79]: @zpawn sorry, this is my mistake - this is actually coming from the settings form itself and not from any persisted settings, please ignore

**Review by AshleyDawson — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (zpawn) [reply src/App.tsx:6]: done

**Pull request merged**